### ISSUE TITLE
HTML5 placeholder support?

### ISSUE THREAD

**Issue by xied75** [jdorn/json-editor#396]
Something like:

```
{
  "format": "placeholder"
}
```

**Comment by dakshika** [comment]
I don't agree on above suggestion. Usually and formatted filed can have placeholder text, so better to use default value in schema as placeholder.

**Comment by mbrodala** [comment]
The default value should be used as regular value just like right now though. To stay close to the JSON schema specs, this could be put in `options`. If you extend the default `string` editor, you can add this locally.

**Comment by xied75** [comment]
@dakshika Default value and placeholder are distinct concept and purpose, therefore those too should NOT be mixed.

@mbrodala After some reading on JsonSchema rfc now I think using format is not a good idea at all. Do you mind point me the relevant paragraph in the spec regarding **options**?

**Comment by mbrodala** [comment]
@xied75 **options** is not part of JSON schema (validation) but a custom property of the JSON editor. This is already used e.g. to provide `enum_titles`.

**Comment by xied75** [comment]
@mbrodala Thanks for reply. As related, I kind of think to extend the "format" as the current implementation is probably not a good idea, because "format" actually is part of JSON schema RFC, and some validators will raise error while encountering any of the non-default (i.e. ipv4, ipv6, date-time, etc.) 
https://github.com/zaggino/z-schema/issues/121

I wonder should we move all these "format" to something else, e.g. all these html input types.

**Comment by coderofsalvation** [comment]
Lets not waste our breath too much on this, as long as 'placeholder' is not allowed by the maintainer, I use this after an empty form got rendered:
```
    $('input').each(>(k, v){
      v.setAttribute( 'placeholder', v.value )
      v.value = ''
    })
```
